File: Problems/125.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        std::string valid = "";


        for(int i = 0; i < s.size(); i++ ){

            if(isalnum(s[i])){
                valid += tolower(s[i]);
            } 
        }

        for(int i = 0; i < valid.size() / 2; i++){
            if (valid[i] != valid[valid.size() - 1 - i]) {
                return false;
            }
        }

        return true;
    }
};

File: Problems/125.rs
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let new_string = s
            .chars()
            .filter(|x| x.is_alphanumeric())
            .collect::<String>()
            .to_ascii_lowercase();
        let new_string = new_string.as_bytes();

        if new_string.len() == 0 {
            return true;
        }

        let mut i = 0;
        let mut j = new_string.len() - 1;

        while i < j {
            match new_string[i].cmp(&new_string[j]) {
                std::cmp::Ordering::Less | std::cmp::Ordering::Greater => return false,
                _ => {
                    i += 1;
                    j -= 1;
                }
            }
        }

        true
    }
}
